chain/client/src/sync/state.rs

```diff
@@ -747,7 +749,7 @@ impl StateSync {
                     // No header found.
                     // It may happen because requested node couldn't build state response.
                     if !header_download.done {
-                        tracing::info!(target: "sync", %shard_id, %hash, "state_response doesn't have header, should be re-requested");
```

Comment:
I think for headers even from GCS they are applied by the client actor because it happens on the rx side of the message queue: https://github.com/near/nearcore/blob/b7c451a2571b4b0f4ba52d8f4da98a0faf0a4fcd/chain/client/src/sync/state.rs#L367-L368

Overall this is horribly messy but I'm trying to limit the blast radius of this fix given that Robin already has a complete rewrite in the pipeline ([#12172](https://github.com/near/nearcore/issues/12172)).

Reply:
oh yeah you're right actually